consensus/state.go

```diff
@@ -1306,6 +1306,22 @@ func (cs *ConsensusState) addProposalBlockPart(height int64, part *types.Part, v
 		}
 		// NOTE: it's possible to receive complete proposal blocks for future rounds without having the proposal
```

Comment:
Had to re-derive why this is the case... I believe it's because of consensus catch-up, where we give nodes lagging in height, the block-parts and precommit votes, which lets the laggard skip all the normal round-syncing and be convinced to progress to the next height.  (There's some spam prevention built in here, btw.)